src/hooks/useDynamicOptionsLoaderState.tsx

```diff
@@ -0,0 +1,333 @@
+  )
+
+  const [freshdeskFieldsState] = useLoadDataState(loadFreshdeskFields)
+
+  const loadFormElementOptionsSets = React.useCallback(
+    async (abortSignal) => {
+      const formElementOptionsSets =
+        await formService.getFormElementOptionsSets(
+          form.organisationId,
+          abortSignal,
+        )
+      return formElementOptionsSets.map<OptionsSetResult>(
+        (formElementOptionsSet) => ({
+          formElementOptionsSet,
+        }),
+      )
+    },
+    [form.organisationId],
+  )
+
+  const [optionsSetResultsState, , setOptionsSetResults] = useLoadDataState(
+    loadFormElementOptionsSets,
+  )
+
+  const optionsSetResults = React.useMemo(() => {
+    if (optionsSetResultsState.status === 'SUCCESS') {
+      return optionsSetResultsState.result
+    }
+    return []
+  }, [optionsSetResultsState])
+
+  const error = React.useMemo<OneBlinkAppsError | undefined>(() => {
+    if (freshdeskFieldsState.status === 'ERROR') {
+      return freshdeskFieldsState.error instanceof OneBlinkAppsError
+        ? freshdeskFieldsState.error
+        : new OneBlinkAppsError('An unknown error has occurred', {
+            originalError: freshdeskFieldsState.error,
+          })
+    }
+    if (optionsSetResultsState.status === 'ERROR') {
+      return optionsSetResultsState.error instanceof OneBlinkAppsError
+        ? optionsSetResultsState.error
+        : new OneBlinkAppsError('An unknown error has occurred', {
+            originalError: optionsSetResultsState.error,
+          })
+    }
+    for (const optionsSetResult of optionsSetResults) {
+      if (optionsSetResult.result?.type === 'ERROR') {
+        return optionsSetResult.result.error
+      }
+    }
+  }, [freshdeskFieldsState, optionsSetResults, optionsSetResultsState])
+
+  if (error) {
+    return (
+      <>
+        <div className="has-text-centered">
+          <i className="material-icons has-text-warning icon-x-large">error</i>
+          <h3 className="title is-3">{error.title}</h3>
+          <p>{error.message}</p>
+          <p className="has-text-grey">
+            {localisationService.formatDatetimeLong(new Date())}
+          </p>
+        </div>
+
+        <OneBlinkAppsErrorOriginalMessage error={error.originalError} />
+      </>
+    )
+  }
+
+  return (
+    <>
+      {optionsSetResults.map((optionsSetResult) => (
+        <React.Fragment key={optionsSetResult.formElementOptionsSet.id}>
+          <LoadOptionsSet
+            optionsSetResult={optionsSetResult}
+            form={form}
+            setOptionsSetResults={setOptionsSetResults}
+          />
+        </React.Fragment>
+      ))}
+      <FormElementOptionsContext.Provider value={optionsSetResults}>
+        <FreshdeskFieldsStateContext.Provider value={freshdeskFieldsState}>
+          {children}
+        </FreshdeskFieldsStateContext.Provider>
+      </FormElementOptionsContext.Provider>
+    </>
+  )
+}
+
+const LoadOptionsSet = React.memo(function LoadOptionsSet({
```

Comment:
What's the point of this one being a component?

Reply:
so the effect can have the individual OptionSetResult as a dep instead of the entire array